### PR TITLE
Add support for SOAP basic client

### PR DESCRIPTION
**Pull request by MadhukaHarith92**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/4618
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina/soap.bal [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import ballerina/http;
+import ballerina/mime;
+
+# Object for the basic SOAP client endpoint.
+public client class BasicClient {
+
+    http:Client soapClient;
+
+    public function init(string url, http:ClientConfiguration? config = ()) returns error? {
```
Comment (ayeshLK) [inline ballerina/soap.bal:25]:
```suggestion
    public function init(string url, *http:ClientConfiguration config) returns error? {
```

File: ballerina/soap.bal [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/mime;
+
+# Object for the basic SOAP client endpoint.
+public client class BasicClient {
```
Comment (TharmiganK) [inline ballerina/soap.bal:21]: Are we supporting SOAP 1.1 and SOAP 1.2 through this client? I don't see a configuration for that

Comment (MadhukaHarith92) [reply ballerina/soap.bal:21]: As per the design[[1]](https://github.com/ballerina-platform/ballerina-standard-library/issues/4618#issuecomment-1639234830), the basic client only supports soap 1.1.

[1] https://github.com/ballerina-platform/ballerina-standard-library/issues/4618#issuecomment-1639234830

**Comment by MadhukaHarith92** [comment]
Closing in favor of  https://github.com/ballerina-platform/module-ballerina-soap/pull/82 with some design changes.

File: ballerina/soap.bal (hunk 2 above)
Comment (MadhukaHarith92) [reply ballerina/soap.bal:21]: Updated in https://github.com/ballerina-platform/module-ballerina-soap/pull/82

**Pull request closed without merging**